pandas/core/reshape/pivot.py

```diff
@@ -557,7 +557,12 @@ def _all_key(key):
                 table_pieces.append(piece)
                 margin_keys.append(all_key)
         else:
-            from pandas import DataFrame
+            margin = (
+                data[cols[:1] + values]
```

Comment:
Why `cols[:1]` here?

Reply:
This is to match the `table.groupby(level=0)` below.

Reply:
Ah, makes sense, thanks!